### PR TITLE
Redesign search page for mobiles

### PR DESCRIPTION
**Pull request by raonismaneoto**
**Feature/Bug description:**
The previous search page was out of the designer's pattern.

screenshot from 2018-11-09 17-32-17

screenshot from 2018-11-09 17-32-27

**Solution:** 
Creating a new html block to improve the search page view on mobile screen.

s5

screenshot from 2018-11-09 17-30-21

screenshot from 2018-11-09 17-31-07

dialog

**TODO/FIXME:** n/a

### CODE REVIEW

File: frontend/search/search.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<md-content layout="row" layout-align="center" class="custom-scrollbar">
+<md-content layout="row" layout-align="center" class="custom-scrollbar" ng-if="!searchCtrl.isMobileScreen()">
```
Comment (JuliePessoa) [inline frontend/search/search.html]: não usa ng-if.
Cria um html separado.
Lá no app.js, onde define o state, utiliza uma função pra definir o html correto.

File: frontend/search/searchController.js [hunk 2]
```diff
@@ -36,18 +36,22 @@
             searchCtrl.institutions = [];
         };
 
-        searchCtrl.search = function search() {
+        searchCtrl.search = function search(ev) {
```
Comment (JuliePessoa) [inline frontend/search/searchController.js]: jsdoc

File: frontend/search/searchController.js [hunk 3]
```diff
@@ -36,18 +36,22 @@
...
             if (searchCtrl.search_keyword) {
-                searchCtrl.makeSearch(searchCtrl.search_keyword, 'institution');
+                searchCtrl.makeSearch(searchCtrl.search_keyword, 'institution').then(() => {
```
Comment (JuliePessoa) [inline frontend/search/searchController.js]: add returns

File: frontend/search/searchController.js [hunk 4]
```diff
@@ -36,18 +36,22 @@
...
+                    if (Utils.isMobileScreen()) {
+                        searchCtrl.showSearchFromMobile(ev);
+                    }
+                });
                 refreshPreviousKeyword();
```
Comment (JuliePessoa) [inline frontend/search/searchController.js]: is there any problem with this function being "parallel" with `makeSearch`?

File: frontend/search/searchController.js [hunk 5]
```diff
@@ -36,18 +36,22 @@
...
             }
         };
 
-        searchCtrl.goToInstitution = function goToInstitution(institutionId) {
+        searchCtrl.goToInstitution = function goToInstitution(institutionId, cameFromDialog) {
```
Comment (JuliePessoa) [inline frontend/search/searchController.js]: create a function `gotoinstituition` with another name, to call in the dialog.
Inside do `md.cancel` and then call this `gotoinstitution`function.

File: frontend/search/searchController.js [hunk 6]
```diff
@@ -58,6 +62,51 @@
             }
         };
 
+        /**
+         * Open up a mdDialog to show the search result in a mobile fone
+         * @param {Object} ev
```
Comment (JuliePessoa) [inline frontend/search/searchController.js]: what type of event?
Use {Event} if its is a js event object.

File: frontend/search/searchController.js (hunk 4 above)
Comment (raonismaneoto) [reply frontend/search/searchController.js]: you mean put it inside the then block?

File: frontend/search/search.css [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    color: #009688;
+}
+
+.search-mobile-action-buttons {
+  margin-right: 6%; 
+  margin-top: 10%;
+}
+
+#search-list-item {
+  height: 50px;
```
Comment (Luiz-FS) [inline frontend/search/search.css]: Use em

File: frontend/search/search.css [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+#search-list-item div div {
+  height: 47px;
```
Comment (Luiz-FS) [inline frontend/search/search.css]: Same for all

**Pull request merged**